src/WireMock.Net/Owin/OwinSelfHost.cs

```diff
@@ -58,37 +65,48 @@ public Task StopAsync()
         private void StartServers()
         {
 #if NET46
-            Console.WriteLine("WireMock.Net server using .net 4.6.x or higher");
+            _logger.Info("WireMock.Net server using .net 4.6.x or higher");
 #else
-            Console.WriteLine("WireMock.Net server using .net 4.5.x or higher");
+            _logger.Info("WireMock.Net server using .net 4.5.x or higher");
 #endif
-
-            Action<IAppBuilder> startup = app =>
-            {
-                app.Use<GlobalExceptionMiddleware>(_options);
-                _options.PreWireMockMiddlewareInit?.Invoke(app);
-                app.Use<WireMockMiddleware>(_options);
-                _options.PostWireMockMiddlewareInit?.Invoke(app);
-            };
-
-            var servers = new List<IDisposable>();
-            foreach (var url in Urls)
+            try
             {
-                servers.Add(WebApp.Start(url, startup));
+                Action<IAppBuilder> startup = app =>
+                {
+                    app.Use<GlobalExceptionMiddleware>(_options);
+                    _options.PreWireMockMiddlewareInit?.Invoke(app);
+                    app.Use<WireMockMiddleware>(_options);
+                    _options.PostWireMockMiddlewareInit?.Invoke(app);
+                };
+
+                var servers = new List<IDisposable>();
+                foreach (var url in Urls)
+                {
+                    servers.Add(WebApp.Start(url, startup));
+                }
+
+                IsStarted = true;
+
+                // WaitHandle is signaled when the token is cancelled,
+                // which will be more efficent than Thread.Sleep in while loop
+                _cts.Token.WaitHandle.WaitOne();
+
+                foreach (var server in servers)
+                {
+                    server.Dispose();
+                }
             }
-
-            IsStarted = true;
-
-            while (!_cts.IsCancellationRequested)
+            catch (Exception e)
             {
-                Thread.Sleep(30000);
+                // Expose exception of starting host, otherwise it's hard to be troubleshooting if keeping quiet
+                // For example, WebApp.Start will fail with System.MissingMemberException if Microsoft.Owin.Host.HttpListener.dll is being located
+                // https://stackoverflow.com/questions/25090211/owin-httplistener-not-located/31369857
+                _runningException = e;
+                _logger.Error(e.ToString());
             }
-
-            IsStarted = false;
-
-            foreach (var server in servers)
+            finally
             {
-                server.Dispose();
+                IsStarted = false;
```

Comment:
foreach (var server in servers) is moved into finally block. The reason of moving disposing server in finally is to make sure allocated resources(like handles in kernel mode) to be cleaned up immediately rather than wait until application is terminated.

// Dispose all servers in finally block to make sure clean up allocated resource on error happening
 servers.ForEach((s) => s.Dispose());